app/models/spree/order_decorator.rb

```diff
@@ -305,12 +305,14 @@ def total_tax
   end
 
   def tax_adjustments
-    adjustments.with_tax +
-      line_items.includes(:adjustments).map { |li| li.adjustments.with_tax }.flatten
+    adjustments + price_adjustments
```

Comment:
if you want to include negatives you can change the scope to include negatives
scope :with_tax,       -> { where('spree_adjustments.included_tax > 0') }

but here you are just including all adjustments, including the ones with zero tax. that means the method become non-sense where tax_adjustments returns all adjustments.

I think you want to keep the method as tax_adjustment but exclude included_tax ZERO.

Reply:
Yes, you are right, I was not thinking of this case, and it will also solve the comment you put above for `adjustment.included_tax.zero?` control. I will fix this, need to cross-check side effect related to this scope use.